src/aatoolbox/datasources/usgs/usgs_ndvi.py

```diff
@@ -0,0 +1,851 @@
+                url_time_str,
+                # Fri, 27 Mar 2015 08:05:42 GMT
+                "%a, %d %b %Y %X %Z",
+            )
+            if not clobber and url_time <= file_time:
+                logger.info(
+                    f"File {filepath} exists, clobber set to False, and "
+                    "file not been modified since last download. "
+                    "Using existing files."
+                )
+                return filepath
+
+        # open file within memory
+        zf = ZipFile(BytesIO(resp.read()))
+
+        # extract single .tif file from .zip
+        for file in zf.infolist():
+            if file.filename.endswith(".tif"):
+                # rename the file to standardize to name of zip
+                file.filename = f"{local_filename}.tif"
+                zf.extract(file, self._raw_base_dir)
+
+        resp.close()
+        return filepath
+
+    def _process(
+        self,
+        clobber: bool,
+        gdf: gpd.GeoDataFrame,
+        feature_col: str,
+        dates_to_process: list,
+        stats_list: Optional[List[str]],
+        percentile_list: Optional[List[str]],
+        kwargs,
+    ) -> Path:
+        """Process data for particular statistic."""
+        # get processed path for particular statistic
+        if stats_list:
+            stat = stats_list[0]
+        elif percentile_list:
+            stat = f"{percentile_list[0]}quant"
+        processed_path = self._get_processed_path(
+            feature_col=feature_col, stat=stat
+        )
+
+        if processed_path.is_file():
+            dates_to_process, df = self._determine_process_dates(
+                clobber=clobber,
+                filepath=processed_path,
+                dates_to_process=dates_to_process,
+            )
+
+            if not dates_to_process:
+                logger.info(
+                    (
+                        "No new {stat} data to process between "
+                        f"{self._start_year}, "
+                        f"dekad {self._start_dekad} "
+                        f"and {self._end_year}, "
+                        f"dekad {self._end_dekad}, "
+                        "set `clobber = True` to re-process this data."
+                    )
+                )
+                return processed_path
+        else:
+            df = pd.DataFrame()
+
+        # process data for necessary dates
+        data = [df]
+        for process_date in dates_to_process:
+            da = self.load_raster(process_date)
+            stats = da.aat.compute_raster_stats(
+                gdf=gdf,
+                feature_col=feature_col,
+                stats_list=stats_list,
+                percentile_list=percentile_list,
+                **kwargs,
+            )
+            data.append(stats)
+
+        # join data together and sort
+        df = pd.concat(data)
+        df.sort_values(by="date", inplace=True)
+        df.reset_index(inplace=True, drop=True)
```

Comment:
You can chain these bad boys if you take out `inplace=True`. I thought you would never miss such an opportunity!

Reply:
My god. Of course.